src/lib/ll_schedule.c

```diff
@@ -374,6 +362,21 @@ static void ll_notify(int message, void *data, void *event_data)
 	spin_unlock_irq(&queue->lock, flags);
 }
 
+static inline void insert_work_to_queue(struct task *w,
+					struct list_item *q_list)
+{
+	struct task *ll_task;
+	struct list_item *wlist;
+
+	/* works are adding to queue in order */
+	list_for_item(wlist, q_list) {
+		ll_task = container_of(wlist, struct task, list);
+		if (w->priority >= ll_task->priority) {
+			list_add_item_before_ref_item(&ll_task->list, &w->list);
```

Comment:
Shouldn't we break the loop after adding?

Reply:
Yes, it was mistake.